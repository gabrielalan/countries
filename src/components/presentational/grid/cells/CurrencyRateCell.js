import React from 'react';
import PropTypes from 'prop-types';

const locale = navigator.language || navigator.userLanguage

export default function CurrencyRateCell({ rates, children }) {
  const link = !rates[children]
    ? "https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html"
    : `https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/eurofxref-graph-${children.toLowerCase()}.en.html`;

  const rate = rates[children] ? `${rates[children].toLocaleString(locale)}` : '--';

  return <div className="currency-rate">
    {children} 
    <a href={link}
      target="blank">
      ({rate})
    </a>
  </div>;
}

CurrencyRateCell.propTypes = {
  children: PropTypes.string.isRequired,
  rates: PropTypes.object.isRequired
};