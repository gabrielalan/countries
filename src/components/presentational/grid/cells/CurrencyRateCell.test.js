import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyRateCell from './CurrencyRateCell';

describe('CurrencyRateCell', () => {
  it('should render the rate together with the currency', () => {
    const tree = renderer.create(<CurrencyRateCell rates={{ USD: 2 }}>USD</CurrencyRateCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render (--) when there is no rate for currency', () => {
    const tree = renderer.create(<CurrencyRateCell rates={{}}>USD</CurrencyRateCell>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
