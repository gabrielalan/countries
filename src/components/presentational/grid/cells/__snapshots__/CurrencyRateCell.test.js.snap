// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyRateCell should render (--) when there is no rate for currency 1`] = `
<div
  className="currency-rate"
>
  USD
  <a
    href="https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html"
    target="blank"
  >
    (
    --
    )
  </a>
</div>
`;

exports[`CurrencyRateCell should render the rate together with the currency 1`] = `
<div
  className="currency-rate"
>
  USD
  <a
    href="https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/eurofxref-graph-usd.en.html"
    target="blank"
  >
    (
    2
    )
  </a>
</div>
`;
